pkg/openstack/instancesv2.go

```diff
@@ -79,6 +79,18 @@ func (os *OpenStack) instancesv2() (*InstancesV2, bool) {
 
 // InstanceExists indicates whether a given node exists according to the cloud provider
 func (i *InstancesV2) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+	if i.regionProviderID {
+		if node.Spec.ProviderID == "" {
+			klog.V(4).Infof("Instance %s should initialized first", node.Name)
+			return true, nil
+		}
+
+		if instanceNodeUnmanaged(node.Spec.ProviderID) {
+			klog.V(4).Infof("Instance %s is not an OpenStack instance", node.Name)
+			return true, nil
```

Comment:
Again, is `true` the correct return value here? I'd have to check the call-sites, but my feeling is that this could lead to incorrect behaviour possibly today, but if not in the future. My feeling is that we should return an error here.